### DISCUSSION TITLE
Updating reference fields

### DISCUSSION THREAD

**Discussion by longj08** [sadieblue/agilityblue.workspace-scripts#31] [Q&A] [answered]
Hi,
After I created a new task in a project and would like to update the reference fields.  Below is my codes to try to add/update a reference field value but I got an error that says "Method invocation failed because [AgilityBlue.ViewModels.Workspace.TaskFormFieldReferenceObjectViewModel] does not contain a method named 'op_Addition'."  Do you know what is the cause on the error or/and do you have a sample script that updates the reference fields?  Thank you!

$task = Get-Task $taskId -IncludeForms

$billingEntriesRef = , @(
                        @{ objectId = 9; key = 1; value = "#1 - Thumb Drive" }
                ) | ConvertTo-Json -Compress
                
$billingEntryToCreate = @{
            Fields = @(
                @{
                    FieldId = 2100
		            Label = "Billing Entries"
		            
		              ReferenceObject = @{
			            Values = @(
				            @{
				                KeyAsString = $billingEntriesRef.ToString()   
				                KeyAsInteger = 1
				            }
				        )
				      }
                   
                }
           )
        }
      
$refTask =$task.forms.sections.fields | where-object {$_.label -eq "Billing Entries"}
      
$refTask.ReferenceObject += $billingEntryToCreate
      
$createdEntries = Set-Task -Entry $task

**Comment by bmward80** [comment, marked as answer]
Hi @longj08,

The problem here is that you're creating and using a JSON string for the "KeyAsString" property. Task form reference fields don't use JSON for references, they use KeyAs* and the key of the object you're attaching, kind of like what you're doing, but you only use one of the KeyAs* fields (depends on the object id), and a "Value" property that represents the display value of the thing you're referencing.

Here is the structure you should have of the `ReferenceObject` property:

```powershell
ReferenceObject = @{
  Values = @(
    @{
      KeyAsInteger = 1
      Value       = "#1 - Thumb Drive"
    }
  )
}
```

Check out the "ReferenceObject" section in the help docs [https://help.agilityblue.com/docs/scripting-tutorials-custom-fields-deep-dive#data-type-reference](https://help.agilityblue.com/docs/scripting-tutorials-custom-fields-deep-dive#data-type-reference) and let me know if you need any further help.

Brandon

**Comment by longj08** [comment]
It works.  Thank you so much Brandon.
